rectools/models/pure_svd.py

```diff
@@ -78,6 +93,15 @@ def __init__(
         self.tol = tol
         self.maxiter = maxiter
         self.random_state = random_state
+        if use_gpu:  # pragma: no cover
+            if not HAS_CUDA:
+                warnings.warn("Forced to use CPU. CUDA is not available.")
+                use_gpu = False
+            if cp is None:
+                warnings.warn("Forced to use CPU. CuPy is not available.")
+                use_gpu = False
+
+        self.use_gpu = use_gpu
```

Comment:
We need to preserve the initial value for config. You can use the idea from [here](https://github.com/MobileTeleSystems/RecTools/blob/main/rectools/models/lightfm.py#L152)

Reply:
Fixed 6c060fbf6f266e9c7347c28431d7c3ae144e6ffa